pkg/controller/common.go

```diff
@@ -64,6 +70,48 @@ func NewController(ctx context.Context, base controllers.Controller, rm Resource
 func (c *Controller) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
 	c.Client = mgr.GetClient()
 	c.Scheme = mgr.GetScheme()
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+
+	preferredResources, err := discoveryClient.ServerPreferredResources()
+	if err != nil {
+		return err
+	}
+
+	deletableResources := discovery.FilteredBy(discovery.ResourcePredicateFunc(func(groupVersion string, r *metav1.APIResource) bool {
+		check := sets.NewString([]string(r.Verbs)...).HasAll("delete", "list", "create")
+		check = check && r.Namespaced
+
+		return check
+	}), preferredResources)
+
+	c.deletableGVKs = make(map[schema.GroupVersionKind]struct{})
+
+	for _, rl := range deletableResources {
+		gv, err := schema.ParseGroupVersion(rl.GroupVersion)
+		if err != nil {
+			return err
+		}
+
+		for i := range rl.APIResources {
+			sar := &kauthn.SelfSubjectAccessReview{
+				Spec: kauthn.SelfSubjectAccessReviewSpec{
+					ResourceAttributes: &kauthn.ResourceAttributes{
+						Verb:     "delete",
+						Group:    gv.Group,
+						Version:  gv.Version,
+						Resource: rl.APIResources[i].Name,
+					},
+				},
+			}
+			if err := c.Client.Create(ctx, sar); err != nil {
+				return err
+			}
+
+			if sar.Status.Allowed {
+				c.deletableGVKs[gv.WithKind(rl.APIResources[i].Kind)] = struct{}{}
+			}
```

Comment:
how about logging exceptions with `else`

Reply:
should not be logging exceptions, the effect here is similar to filter